### PR TITLE
Skip non-numeric values in image trace plot 

### PR DESCRIPTION
**Pull request by archmoj**
Fix in respect to @emmanuelle 's comment
https://github.com/plotly/plotly.js/issues/4301#issuecomment-548103964
i.e. by drawing those pixels with full transparency.

[before vs after demo](https://codepen.io/MojtabaSamimi/pen/oNNoMBE?editors=0010)

@alexcjohnson 

cc: @nicolaskruchten 
cc: @plotly/plotly_js

### CODE REVIEW

File: test/image/mocks/image_non_numeric.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "data": [{
+    "type": "image",
+    "hoverinfo": "all",
+    "z": [
+      [
+        [true, true, true],
+        [false, false, false]
```
Comment (alexcjohnson) [inline test/image/mocks/image_non_numeric.json]: True, there's something nice about ensuring we see the right axis ranges even when *all* of the pixels are invalid. But perhaps then a second mock (or a second subplot on the same one?) that includes a couple of valid pixels as well as various kinds of invalid pixels - any single channel missing for example.

Comment (archmoj) [reply test/image/mocks/image_non_numeric.json]: Good call. That mock should be improved.

Comment (archmoj) [reply test/image/mocks/image_non_numeric.json]: Done in https://github.com/plotly/plotly.js/pull/4325/commits/d6abde903a8988edc218b63498381d9d2fa472a5.

**Comment by etpinard** [comment]
Nice PR @archmoj - can I why you made some of the colorscale attributes `editType: 'calc'` now? I suspect this was to make the plot step faster, but I'd like to confirm with you.

**Comment by archmoj** [comment]
@etpinard Yes. That was for speed and ( if I recall ) not for calling `isNumeric` during the `plot`.

**Comment by etpinard** [comment]
> Yes. That was for speed and ( if I recall ) not for calling `isNumeric` during the `plot`.

Yeah, that very nice. Most colorscale computations for other traces are done during the calc step, so this is fantastic :+1:

**Pull request merged**